provider/disk_utils.py

```diff
@@ -0,0 +1,97 @@
+"""
+Disk utilities function for managing disks and filesystems in guest environments.
+
+This module contains a function related to disk write command execution on
+different OS types.
+"""
+
+from virttest import utils_disk
+from virttest.utils_misc import get_linux_drive_path
+from virttest.utils_windows.drive import get_disk_props_by_serial_number
+
+
+def execute_io_test(
+    test,
+    params,
+    vm,
+    image,
+    fstype,
+    dst_dir=None,
+    io_command=None,
+    timeout=60,
+):
+    """
+    Execute the dd write test on the disk image inside the guest.
+    Note: The target disk image should be the uninitialized disk.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param image: The image name. e.g: stg0
+    :param dst_dir: The destination directory for I/O write testing.
+    :param io_options: The options for I/O command.
+    :param timeout: The timeout for the I/O command.
+    """
+
+    if fstype not in ("ntfs", "xfs"):
+        test.error("The fstype is not supported, only 'xfs' and 'ntfs' are")
```

Comment:
Hi @mcasquer
As @qingwangrh mentioned,  He does not need to care about the `fstype` in the general scenarios, so I think we could provide a default for the users if `fstype` is `None`.

Reply:
Updated, assigning `xfs` by default if `fstype` is `None`